### PR TITLE
extra component specs & component file renaming (index.js -> Class.js, index.js imports Class)

### PR DESCRIPTION
**Pull request by jacogr**
For now, for discussion only, basically investigating while I delve through and get familiar.... 

Basically I started looking at the components and expanding the specs/unit tests. Started with probably one of the most "ummm, ok" components, which only wraps AutoComplete. However, even this did yield some results in just testing the individual rendering -

1. structure in-place for doing this type of component-level testing (one level below E2E), i.e. via jsdom as is the React standard
2. existing tests still in-place, i.e. localStorage was slightly tweaked to fit in now that jsdom provides a window object
3. even with this simple component, expanded the required propTypes to not have React run-time errors/warnings (there is an issue related to this logged, easier to evaulate when picked up as warnings at unit-test time)
4. Slightly changes the structure of the component where the files are named appropriately, i.e. index.js -> AutoComplete.js, index.js now exports AutoComplete (slight indirection, however material-ui also uses this approach, helps if you know which file you work with - sadly this looks like a new file instead of a simple rename, but it happens)

In short, very basic rendering test in this case (others will be more relevant), however some of the React warnings can be addressed without looking at the console, and if we do break in some way, shape or form with the wrapping, we don't need to get to E2E time...

Valid approach?

### CODE REVIEW

**Comment by goldylucks** [comment]
1. we decided not to do any react specific because:
a. if we move to another framework they become irrelevant
b. e2e + UT would b enough
4. seems like extra work without much gain to me:)
if i c `import AutoComplete from 'components/AutoComplete'` I know its a ref to the index file

**Comment by jacogr** [comment]
Sadly (since I'm currently exactly in the process of swapping frameworks, via an "upgrade" from Angular 1.5 to 2.0), E2E tests also become useless in these "other framework" cases. It still breaks and you need to re-do/change/expand the tests anyway. While tests are never an or-or, rather an and-and (expansive unit tests & expansive end-to-end tests) providing benefits if you can do both - but as you rightly pointed out, any code requires time and effort, neither are free.

The idea is to change the import line, but rather to put the main login in an appropriately-named file. Small indirection from index.js (doesn't affect imports), however when you have multiples open of the core meat at any time it does help in discoverability. (Really not that much of an issue with the current size, but is nice with minimal effort.)

Like I mentioned, exactly opened this for the purposes of discussion, so the extra insight is very valuable :)

**Comment by goldylucks** [comment]
discoverability - u mean like in your text editor / IDE? instead of seeing Comp1/index.js Comp2/index.js in the tabs u will c Comp1.js Comp2.js?

yea I think u right. it is worth it :)

**Comment by jacogr** [comment]
Yes, in the editor. I didn't actually think of it until I saw that material-ui.com did it that way, slightly better clarity in what the file does and the component imports are still easy. I always tend to end up with way too many tabs, gaining some extra clarity with little effort is always appreciated, at least for me. (Obviously it is editor specific, i.e. in my case it is 'index.js - Box', 'index.js - Footer', etc. It can and does get slightly messy at times.) 

I really don't have strong feelings on this one, obviously like the idea, but what we have works as well.

**Comment by tomusdrw** [comment]
1. It's always good to have testing framework in place (it encourages you to write the test by lowering the mental cost you need to take (copy&paste:))). I would like to have it, but wouldn't make fully unit testing the component a strict requirement (the border between good unit test and implementation specific unit test is very thin in such cases, the latter being difficult to maintain).

2. Totally agree about component structure. We adopted `tj/boilerplate-frontend` approach, but I also prefer more meaningful filenames (editor tabs, stack traces (now everything is crashing in `index.js:<someline>`), search).

So in summary: :+1: great work. Let's get that merged as soon as possible :)

**Comment by jacogr** [comment]
Agreed on 1, i.e. it shouldn't be a strict requirement - we don't want to end up writing tests for the sake of writing tests. (I love them when they are there, but everything needs to add value.)

What I will do in light of 2 (since it seems that Adam is in slight agreement) is to expand this PR and do all the renames for client/components (trying to keep the PR sizes manageable, the renames only will make a proper review possible).

**Comment by goldylucks** [comment]
Adam is in total agreement!
Mental cpu is a precious resource :)

File: .eslintrc.json [hunk 1]
```diff
@@ -1,4 +1,13 @@
...
+  "parser": "babel-eslint",
+  "globals": {
+    "after": true,
+    "afterEach": true,
+    "before": true,
+    "beforeEach": true,
+    "describe": true,
+    "expect": true,
+    "it": true
+  }
```
Comment (goldylucks) [inline .eslintrc.json]: I prefer this approach too than explicitly stating it in each UT.

@tomusdrw I think we talked about it in the last retreat, I don't remember if u had any objections for it?

**Comment by goldylucks** [comment]
@tomusdrw why removed the looks good label?

@jacogr after all the changes did u run testOnce locally?
everything cleared?

**Comment by jacogr** [comment]
I asked @tomusdrw to remove the label - since I did the bulk of the renames after the label was added, I wanted to make sure it gets eyeballs before it gets merged.

Locally both `testOnce` passes (same as on travis) as well as nightwatch (which doesn't run on travis for me)

**Comment by goldylucks** [comment]
ok cool
if everything u did is the import/export refactoring, I think we can merge this

@tomusdrw ?
